packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1732,29 +1732,64 @@ class TransformLayer extends OffsetLayer {
 class OpacityLayer extends OffsetLayer {
   /// Creates an opacity layer.
   ///
-  /// The [alpha] property must be non-null before the compositing phase of
-  /// the pipeline.
+  /// Either the [alpha] or [opacity] property must be non-null before the compositing phase of
+  /// the pipeline. Providing an [alpha] is deprecated and [opacity] should be used
+  /// instead.
   OpacityLayer({
+    @Deprecated(
+      'Prefer using opacity instead of alpha. '
+      'This feature was deprecated after v2.13.0-0.0.pre.'
+    )
     int? alpha,
+    double? opacity,
     super.offset,
-  }) : _alpha = alpha;
+  }) : _opacity = opacity ?? (alpha != null ? _alphaToOpacity(alpha) : null);
```

Comment:
we should assert that we're not passing both (i.e. that at least one of them is null)